install/installer/pkg/config/v1/experimental/experimental.go

```diff
@@ -132,15 +132,21 @@ type WsManagerBridgeConfig struct {
 }
 
 type ServerConfig struct {
-	WorkspaceDefaults                 WorkspaceDefaults `json:"workspaceDefaults"`
-	OAuthServer                       OAuthServer       `json:"oauthServer"`
-	Session                           Session           `json:"session"`
-	GithubApp                         *GithubApp        `json:"githubApp"`
-	ChargebeeSecret                   string            `json:"chargebeeSecret"`
-	DisableDynamicAuthProviderLogin   bool              `json:"disableDynamicAuthProviderLogin"`
-	EnableLocalApp                    *bool             `json:"enableLocalApp"`
-	RunDbDeleter                      *bool             `json:"runDbDeleter"`
-	DefaultBaseImageRegistryWhiteList []string          `json:"defaultBaseImageRegistryWhitelist"`
+	WorkspaceDefaults                 WorkspaceDefaults   `json:"workspaceDefaults"`
+	OAuthServer                       OAuthServer         `json:"oauthServer"`
+	Session                           Session             `json:"session"`
+	GithubApp                         *GithubApp          `json:"githubApp"`
+	ChargebeeSecret                   string              `json:"chargebeeSecret"`
+	DisableDynamicAuthProviderLogin   bool                `json:"disableDynamicAuthProviderLogin"`
+	EnableLocalApp                    *bool               `json:"enableLocalApp"`
+	RunDbDeleter                      *bool               `json:"runDbDeleter"`
+	DefaultBaseImageRegistryWhiteList []string            `json:"defaultBaseImageRegistryWhitelist"`
+	BlockedRepositories               []BlockedRepository `json:"blockedRepositories,omitempty"`
```

Comment:
[nit] Would it make sense to comment the usage of a particular field? I know it's slightly increasing the scope of the PR here but I believe it would make onboarding new people to the team easier if they could read something like:

```suggestion
    // BlockedRepositories define repositories (by regex) which can be blocked.
    // Repositories are often blocked due to abuse or ...
    // To block a repository foo/bar, specify:
    //  { UrlRegExp: "https://github.com/foo/bar", BlockUser: false }
    // ...
	BlockedRepositories               []BlockedRepository `json:"blockedRepositories,omitempty"`
```

This could also act as a good basis for documentation

Reply:
At least in this case, this documentation already exists in the server config:

https://github.com/gitpod-io/gitpod/blob/8ce8f7a5d014b6726245218291ada85f17e9d155/components/server/src/config.ts#L160-L164

and there is a pretty close mapping between fields in this struct and those config fields.

Reply:
Just a heads up, There's #8441 to generate config documentation from the struct itself! Could be done together!